Source/Core/DiscIO/SplitFileBlob.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "DiscIO/SplitFileBlob.h"
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <fmt/format.h>
+
+#include "Common/Assert.h"
+#include "Common/FileUtil.h"
+#include "Common/IOFile.h"
+#include "Common/MsgHandler.h"
+
+namespace DiscIO
+{
+SplitPlainFileReader::SplitPlainFileReader(std::vector<SingleFile> files)
+    : m_files(std::move(files))
+{
+  m_size = 0;
+  for (const auto& f : m_files)
+    m_size += f.size;
+}
+
+std::unique_ptr<SplitPlainFileReader>
+SplitPlainFileReader::Create(const std::string& first_file_path)
+{
+  constexpr std::string_view part0_iso = ".part0.iso";
+  if (!first_file_path.ends_with(part0_iso))
+    return nullptr;
+
+  const std::string_view base_path =
+      std::string_view(first_file_path).substr(0, first_file_path.size() - part0_iso.size());
+  std::vector<SingleFile> files;
+  size_t index = 0;
+  u64 offset = 0;
+  while (true)
+  {
+    File::IOFile f(fmt::format("{}.part{}.iso", base_path, index), "rb");
+    if (!f.IsOpen())
+      break;
+    const u64 size = f.GetSize();
+    if (size == 0)
+      return nullptr;
+    files.emplace_back(SingleFile{std::move(f), offset, size});
+    offset += size;
+    ++index;
+  }
+
+  if (files.size() < 2)
+    return nullptr;
+
+  files.shrink_to_fit();
+  return std::unique_ptr<SplitPlainFileReader>(new SplitPlainFileReader(std::move(files)));
```

Comment:
Any reason you didn't want to use `std::make_unique` here?

Reply:
I can't. The constructor is private, so `make_unique` can't construct the object.

Reply:
Have you considered one of [these designs](https://seanmiddleditch.github.io/enabling-make-unique-with-private-constructors/)?

I'm partial to the `ProtectedTag` or `PrivateTag` approach myself.  Already have used it before in this code base in [BalloonTip](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/DolphinQt/Config/Graphics/BalloonTip.h#L29).

Reply:
The way this PR does it is the way we already do it for the other blob types, so I don't think this needs to be changed in this PR. But feel free to submit this tag approach as a separate PR if you want to.